apps/prairielearn/python/colors.py

```diff
@@ -1,6 +1,9 @@
+import warnings
 from typing import Optional
 
-colors: dict[str, str] = {
+from coloraide import Color
+
+pl_colors: dict[str, str] = {
```

Comment:
@michaelfromyeg Looking at the documentation, I think we can actually extend the `Color` class to allow for more preset colors: https://facelessuser.github.io/coloraide/color/#custom-color-classes

That way, we can get rid of this hardcoded dict altogether!

Reply:
Awesome good catch; will make the change.

Reply:
Hmm... I opened [a discussion item](https://github.com/facelessuser/coloraide/discussions/332) on their repo to see if this is supported; I couldn't see a nice way to override/extend the `__init__` method (which is effectively what this needs). The custom color are more-so around tweaking the color class' core functionality.

Reply:
Ahhh I see. Not a huge deal if we can't get this to work, thanks for following up on this idea!

Reply:
Since I was linked over here.  I'll elaborate in greater detail for interested parties here. Adding more preset colors is certainly achievable. In general, our main focus was to provide compatibility with CSS syntax, but more colors can certainly be added.

Custom `Color` classes are meant to allow a user to register and deregister plugins to get the functionality you need. You can also tweak some defaults, but in general, that is the extent of what is intended. Someone could certainly override core code to potentially do more, but that was not the intention.

The idea is that a user would sub-class the `Color` object to configure defaults and plugins without messing up the base class that may be used by others. Generally, you don't have to change core code, it is mainly plugins. In this project's case, you were using named colors for the sRGB color space, so it makes sense to extend the sRGB plugin to accept a few more custom names.

As you are working in the sRGB color space, and your presets are sRGB colors, this is a sRGB color space issue, so you'd create a custom sRGB color space that will recognize your new custom serialization.

Basically, you'd subclass the sRGB plugin and add a little logic to handle your custom names and then fallback to the normal serialization if custom presets were not found. Afterwards, create a subclass of `Color` and register your custom sRGB plugin with your new custom Color object. Then you just use your new custom `Color` object throughout your code.

In the discussion, I linked a live example that essentially does everything I just mentioned. I'll link it [here](https://facelessuser.github.io/coloraide/playground/?source=https%3A%2F%2Fgist.githubusercontent.com%2Ffacelessuser%2F0b129c1faf7f3f59c0de40eeaaab5691%2Fraw%2Fdc7c87e6468eb3aea38ce7eb24778de9f0778fd9%2Fcustom_color.py#) as well. Just click edit to view the full source.

Hopefully, that makes things a bit more clear.